clar2wasm/src/wasm_generator.rs

```diff
@@ -1913,21 +1949,52 @@ impl ASTVisitor for WasmGenerator<'_> {
         Ok(builder)
     }
 
-    fn visit_nft_get_owner<'b>(
+    fn traverse_nft_get_owner<'b>(
         &mut self,
         mut builder: InstrSeqBuilder<'b>,
         _expr: &SymbolicExpression,
         token: &ClarityName,
-        _identifier: &SymbolicExpression,
+        identifier: &SymbolicExpression,
     ) -> Result<InstrSeqBuilder<'b>, InstrSeqBuilder<'b>> {
-        // Identifier is on the stack, but we need to push the token name onto
-        // the stack, then we can call the host interface function
-        // `nft_get_owner`
+        // Push the token name onto the stack
         let (id_offset, id_length) = self.add_identifier_string_literal(token);
         builder
             .i32_const(id_offset as i32)
             .i32_const(id_length as i32);
 
+        // Push the identifier onto the stack
+        builder = self.traverse_expr(builder, identifier)?;
+
+        let identifier_ty = self
+            .get_expr_type(identifier)
+            .expect("NFT identifier must be typed")
```

Comment:
I think we should rewrite our error handling (in another PR). I don't see why this kind of errors should panic compared to traversal errors which return an `Err`.

Reply:
Yeah, I go back and forth on this. There are some things that should really never happen. Should we never panic in any situation and always return an error? I'm not sure what's best. The existing blockchain code is full of panics when something that shouldn't ever happen happens. But... that has caused problems in the past, so maybe it would be better if there were none.

Reply:
I opened #110 for this.